### PR TITLE
Hologram Projectors for TGC!

### PR DESCRIPTION
**Pull request by NamelessFairy**
## About The Pull Request

Adds a new holodeck layout that features a TGC card fighting arena, complete with holographic representations of your cards. Cards act the same as physical cards when displayed except you can see the stats of the cards without needing to inspect and the cards stats can be modified on the fly for keeping track of equipment.
Example:
image
## Why It's Good For The Game

TGC is a significantly more complicated game then the other ones we have like UNO and CAS and is extremely messy to play on a table ingame, this provides a much clearer way of visualizing the game by having all active creature stats on full display at all times without having to rely on inspecting cards to check.
## Changelog
:cl:
add: Introducing a new holodeck map, the TGC Arena, featuring hologram projectors for your trading cards.
fix: Janitor and Intern TGC cards are now considered creatures rather than just humans.
balance: The price of card packs has been reduced from double a paycheck to 3 quarters of one. 
balance: The number of cards available in the good clean fun vendor has been doubled.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
I wanted to try something similar by making a dual disk or something like that but the cards felt so complicated that I couldn't wrap my head around it
Props to you for figuring out a way to do it

File: code/game/objects/items/tcg/tcg.dm [hunk 1]
```diff
@@ -470,6 +470,15 @@ GLOBAL_LIST_EMPTY(tcgcard_radial_choices)
 	///The rarity of this card, determines how much (or little) it shows up in packs. Rarities are common, uncommon, rare, epic, legendary and misprint.
 	var/rarity = "uber rare to the extreme"
 
+	///Below vars are used for displaying holographic summons
```
Comment (Fikou) [inline code/game/objects/items/tcg/tcg.dm:473]: this isnt an autodoc

Comment (NamelessFairy) [reply code/game/objects/items/tcg/tcg.dm]: Resolving this as this code was removed

**Comment by Troxonater** [comment]
This is really cool. Perhaps we should look into making trading cards persist between rounds for a character for shits and giggles? Trying to get good cards for your deck could be a fun thing to do for assistant mains between antag rolls. BTW I know nothing about the actual trading card game so tell me if this is a dumb idea or somehow already implemented.

**Comment by Wallemations** [comment]
If you're putting the holograms on a dark background, the color for the blue text should be much lighter to contrast better

**Comment by NamelessFairy** [comment]
> If you're putting the holograms on a dark background like the screenshot shows, the color for the blue text should be much lighter to contrast better

I'd changed the resolve color in the commit before your comment, it now looks like this, I was concerned using a brighter color would clash with the AI sprite but this is actually more visible on the AI (although still not "easy" to read unfortunately)

image

**Review by Twaticus — APPROVED** [review]

File: code/game/objects/items/tcg/tcg_machines.dm [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	var/tapped = FALSE
+
+	var/obj/effect/overlay/card_summon/hologram
+	var/obj/effect/overlay/status_display_text/power_overlay
+	var/obj/effect/overlay/status_display_text/resolve_overlay
+	var/power_color = "#af2323"
+	var/resolve_color = "#3492d0"
+	var/modified_color = "#1db327"
+
+	var/team_color = "#77abff"
```
Comment (ArcaneMusic) [inline code/game/objects/items/tcg/tcg_machines.dm:123]: Autodoc these if you would, nothing too crazy

File: code/game/objects/items/tcg/tcg_machines.dm [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+/obj/machinery/trading_card_holder/red
+	summon_offset_y = -1
+	summon_type = /obj/structure/trading_card_summon/red
+
+#define STAT_Y -23
+#define POWER_X -14
+#define RESOLVE_X 12
+#define DEFAULT_POWER_COLOR "#af2323"
+#define DEFAULT_RESOLVE_COLOR "#231ac0"
```
Comment (ArcaneMusic) [inline code/game/objects/items/tcg/tcg_machines.dm:101]: Move to the top of the file for legibility if you would for consistency

**Review by ArcaneMusic — CHANGES_REQUESTED** [review]
Eyes pogging face TCG revival 2023?

**Review by ArcaneMusic — APPROVED** [review]
We agreed on qnty and cost of packs over DMS, It's time to d- d- du-du- DUDUDUDUDUEL

**Pull request merged**